rust/lance-arrow/src/schema.rs

```diff
@@ -50,6 +52,16 @@ impl SchemaExt for Schema {
         Ok(Self::new_with_metadata(fields, self.metadata.clone()))
     }
 
+    fn without_column(&self, column_name: &str) -> Schema {
+        let fields: Vec<FieldRef> = self
+            .fields()
+            .iter()
+            .filter(|f| f.name() != column_name)
+            .cloned()
+            .collect();
+        Self::new_with_metadata(fields, self.metadata.clone())
+    }
```

Comment:
Should we document the contract with field ids (this method leaves them unchanged it appears)?

Reply:
This operates on Arrow schemas, so no implications for field ids.